module/userProjects/helper.js

```diff
@@ -2864,5 +2928,100 @@ function _projectData(data) {
     })
 }
 
+/**
+  * Update Report Information In Solutions.
+  * @method
+  * @name _addReportInformationInSolution 
+  * @param {String} solutionId - solution id.
+  * @param {Object} userProfile - user read response
+  * @returns {Object} Solution information.
+*/
+
+function _addReportInformationInSolution(solutionId = "", userProfile = {}) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if ( userProfile && solutionId ) {
+
+                let district = {};
+                let organisation = [];
+
+                for (const location of userProfile["userLocations"]) {
+                    if ( location.type == CONSTANTS.common.DISTRICT ) {
+                        district["locationId"] = location.id;
+                        district["name"] = location.name;
+                    }
+                }
+
+                for (const org of userProfile["organisations"]) {
+                    let orgData = {};
+                    orgData.orgName = org.orgName;
+                    orgData.organisationId = org.organisationId;
+                    organisation.push(orgData);
+                }
+
+                let solutionDocument = await solutionsQueries.solutionsDocument({
+                    _id: solutionId
+                },
+                    [
+                        "_id",
+                        "reportInformation"
+                    ]
+                );
+                
+                if( !solutionDocument.length > 0 ) {
+                    throw {
+                        message : CONSTANTS.apiResponses.SOLUTION_NOT_FOUND,
+                        status : HTTP_STATUS_CODE['bad_request'].status
+                    }
+                }
+
+                solutionDocument = solutionDocument[0];
+                let reportInformation = solutionDocument.reportInformation ? solutionDocument.reportInformation : {};
+                if ( reportInformation ) {
+                    reportInformation["districts"] = [];
+                    reportInformation["districts"].push(district);
+                    reportInformation["organisations"] = organisation;
+                } else {
+
+                    const checkDistrictExist = reportInformation["districts"].some(eachDistrict => {
+                      if (eachDistrict.locationId == district["locationId"]) {
+                        return true;
+                      }
+                    });
+
+                    if ( !checkDistrictExist ) {
+                        reportInformation["districts"].push(district);
+                    }
+
+                    reportInformation["organisations"].concat(organisation);
+                }
+
+                await solutionsQueries.updateSolutionDocument
```

Comment:
@priyanka-TL we should use addtoset so as to avoid override in case of parallel updates.